flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamingMonitorFunction.java

```diff
@@ -208,6 +210,67 @@ public void testCheckpointRestore() throws Exception {
     }
   }
 
+  @Test
+  public void testConsumeWithMaxSnapshotsPerMonitorInterval() throws Exception {
+    List<List<Record>> recordsList = generateRecordsAndCommitTxn(10);
+
+    final ScanContext scanContext1 = ScanContext.builder()
+        .monitorInterval(Duration.ofMillis(100))
+        .maxSnapshotsPerMonitorInterval(0)
+        .build();
+
+    AssertHelpers.assertThrows("Should throw exception because of invalid config",
+        IllegalArgumentException.class, "must be greater than zero",
+        () -> {
+          createFunction(scanContext1);
+          return null;
+        }
+    );
+
+    final ScanContext scanContext2 = ScanContext.builder()
+        .monitorInterval(Duration.ofMillis(100))
+        .maxSnapshotsPerMonitorInterval(-10)
+        .build();
+
+    AssertHelpers.assertThrows("Should throw exception because of invalid config",
+        IllegalArgumentException.class, "must be greater than zero",
+        () -> {
+          createFunction(scanContext2);
+          return null;
+        }
+    );
+
+    for (int maxSnapshotsNum = 1; maxSnapshotsNum < 15; maxSnapshotsNum = maxSnapshotsNum + 1) {
+      ScanContext scanContext = ScanContext.builder()
+          .monitorInterval(Duration.ofMillis(100))
+          .splitSize(1000L)
+          .maxSnapshotsPerMonitorInterval(maxSnapshotsNum)
+          .build();
+
+      FlinkInputSplit[] expectedSplits = FlinkSplitPlanner
```

Comment:
I think we should directly define the number of expected splits and avoid using `FlinkSplitPlanner`. if `FlinkSplitPlanner` is not honoring  `maxSnapshotsPerMonitorInterval`, the assertion later will still be correct

Reply:
`FlinkSplitPlanner` here is using the same `scanContext` as `StreamingMonitorFunction`, will it still produce different splits?

Reply:
If `FlinkSplitPlanner` didn't honor the `maxSnapshotsPerMonitorInterval` option correctly, this unit test won't be able to detect it. Both the expected and actual splits use the same planner and will generate the same planning result